hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java

```diff
@@ -354,8 +444,13 @@ protected synchronized void addStream(ChunkInfo chunkInfo) {
   }
 
   protected ChunkInputStream createChunkInputStream(ChunkInfo chunkInfo) {
-    return new ChunkInputStream(chunkInfo, blockID,
-        xceiverClientFactory, pipelineRef::get, verifyChecksum, tokenRef::get);
+    if (blockInputStream != null) {
```

Comment:
blockInputStream != null implies the file is opened by SCR.

Reply:
Yeah, add one comment.